src/platform/detection.ts

```diff
@@ -2,6 +2,34 @@ import {Platform} from 'react-native'
 import {getLocales} from 'expo-localization'
 import {dedupArray} from 'lib/functions'
 
+/**
+ * Attempt to ascertain if the agent is a touchscreen device.
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent
+ */
+function hasTouchscreen() {
+  if ('maxTouchPoints' in navigator) {
+    return navigator.maxTouchPoints > 0
+  } else if ('msMaxTouchPoints' in navigator) {
+    // @ts-ignore navigator should exist
+    return navigator.msMaxTouchPoints > 0
+  } else {
+    const mQ = matchMedia?.('(pointer:coarse)')
+    if (mQ?.media === '(pointer:coarse)') {
+      return !!mQ.matches
+    } else if ('orientation' in window) {
+      return true // deprecated, but good fallback
+    } else {
+      // Only as a last resort, fall back to user agent sniffing
```

Comment:
Why is `user-agent` sniffing the last resort?

Reply:
It's mostly unreliable and not considered best practice.

Refs:
https://javascript.plainenglish.io/3-facts-you-should-know-about-user-agent-a476251ee3b5
https://css-tricks.com/browser-detection-is-bad/
https://stackoverflow.com/questions/661213/why-is-browser-sniffing-not-a-recommended-practice
https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent


That being said I've done it in most cross-platform projects I've worked on over the years so...... ya know. Just thought it was worth keeping the comment in there.

Reply:
Appreciate it!